### PR TITLE
Feat/form UI

### LINKED ISSUE #11979: Passphrase: Form UI
Base form UI so we can start working on accessing hidden wallets using this form.

image

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/11979

## Screenshots:

https://github.com/trezor/trezor-suite/assets/36101761/f9f3578f-fb6a-400c-b5fe-e75eeddb8b50

### CODE REVIEW

**Comment by juriczech** [comment]
@SocketSecurity ignore npm/layout-base@1.0.2

File: suite-native/module-passphrase/package.json [hunk 1]
```diff
@@ -20,6 +21,7 @@
         "@suite-native/navigation": "workspace:*",
         "@trezor/styles": "workspace:*",
         "react": "18.2.0",
-        "react-native": "0.73.2"
+        "react-native": "0.73.2",
+        "react-native-reanimated": "3.7.0-nightly-20240101-9d365ae7b"
```
Comment (PeKne) [inline suite-native/module-passphrase/package.json:25]: why is nightly version needed?

File: suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx [hunk 2]
```diff
@@ -1,17 +1,97 @@
...
+}));
+
+const animationWrapperStyle = prepareNativeStyle(() => ({
+    overflow: 'hidden',
+}));
+
 export const PassphraseFormScreen = () => {
+    const { applyStyle, utils } = useNativeStyles();
+
+    const cardHeight = useSharedValue({ height: ALERT_CARD_HEIGHT });
```
Comment (PeKne) [inline suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx]: why is this value an object? Could not it be a number instead?
```suggestion
    const cardHeight = useSharedValue(ALERT_CARD_HEIGHT);
```

File: suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx [hunk 3]
```diff
@@ -1,17 +1,97 @@
...
+                            id="modulePassphrase.subtitle"
+                            values={{
+                                bold: chunks => <Text variant="highlight">{chunks}</Text>,
+                            }}
+                        />
+                    </Text>
+                </VStack>
+                <VStack spacing="medium">
+                    <View style={applyStyle(animationWrapperStyle)}>
+                        <Animated.View style={[animationStyle]}>
```
Comment (PeKne) [inline suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx]:
```suggestion
                        <Animated.View style={animationStyle}>
```

**Review by PeKne — COMMENTED** [review]
The spent time was worth it. The animation looks really neat ⭐️. 

Just few minor questions.

**Comment by PeKne** [comment]
nit: could you be more specific with branch/PR names and commit messages next time? 🙏 Those that you used in this PR are very generic.

File: suite-native/module-passphrase/package.json (hunk 1 above)
Comment (juriczech) [reply suite-native/module-passphrase/package.json:25]: Not needed for this PR, but it's used everywhere so we want to keep same version as everywhere.

Comment (PeKne) [reply suite-native/module-passphrase/package.json:25]: Oh, I wasn't aware of this. 👍

File: suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx (hunk 3 above)
Comment (juriczech) [reply suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx]: Fixed

File: suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/module-passphrase/src/screens/PassphraseFormScreen.tsx]: Fixed

**Review by PeKne — APPROVED** [review]

**Pull request merged**